copier/vcs.py

```diff
@@ -10,15 +10,30 @@
 from packaging import version
 from packaging.version import InvalidVersion, Version
 from plumbum import TF, ProcessExecutionError, colors, local
-from plumbum.cmd import git
 
 from .errors import DirtyLocalWarning, ShallowCloneWarning
 from .tools import TemporaryDirectory
 from .types import OptBool, OptStr, StrOrPath
 
+try:
+    from plumbum.cmd import git
+
+    GIT_VERSION = Version(re.findall(r"\d+\.\d+\.\d+", git("version"))[0])
+except ImportError:
+    import warnings
+
+    warnings.warn(
+        "git command not found, some functionality might not work", UserWarning
+    )
+
+    def git(*args, **kwargs):
```

Comment:
I don't think this will work as expected because plumbum commands can get subscripted, and we actually do that quite often.

Example:

```py
_git = git["-C", dst_path]
_git("clone", src_path)
```

You can do all this inside a method and decorate it with `@proxy_tools.module_property`, so it's lazy loaded. See https://github.com/jtushman/proxy_tools.

Another way would be to also put it inside a method, but don't use that dependency. In that case, every usage of `vcs.git` should become refactored `vcs.git()` (because that method would return the git command). This way we'd save that dependency.

Since you'd be doing import within a lazy method, you could let the `ImportError` raise in case something actually requires git.

Reply:
Thanks @yajo, it's clear that this needs more work. Will give all this some thought!

Reply:
No problem! I marked this as draft in the mean time.

Reply:
I ended up going for your second option (called it `get_git` instead of `git` to minimize confusion, although it makes the diff larger). Let me know what you think.